src/seedu/addressbook/commands/Command.java

```diff
@@ -65,4 +65,9 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+    /**
+     * @return whether the command type is one that mutates data.
+     */
+    public abstract boolean isMutating();
```

Comment:
Try to be more specific on this header comment; the abstract class is a contract that we are providing to the client classes, hence the documentation should be precise. e.g., think what happens when you try to add a duplicate entry to address book. does it still mutate data?